packages/vite/src/node/logger.ts

```diff
@@ -26,6 +26,10 @@ export interface LogErrorOptions extends LogOptions {
   error?: Error | RollupError | null
 }
 
+export const buildFailInfo: Record<string, boolean> = {
+  failed: false,
+}
```

Comment:
I'm not really sure it's safe to rely on a singleton to record the build state, maybe we can move how we record the time to here instead?

https://github.com/vitejs/vite/blob/f9c33add88e8edcd54316629724730d8049447f1/packages/vite/src/node/build.ts#L698-L717

We can start the timestamp before `rollup()`, and end it after `res` is finished. It might also be nice to show "Build failed in XXXms" in the catch handler so that the user knows how long it took so far.

Reply:
Show the failed build time looks good. 👍 

However, at the same time, we should also stop the completion time of the output of the `reporter` plugin.

Reply:
Yeah I was thinking that if we record the time in the code I linked. Then the reporter shouldn't be handling the logging for the built time anymore.

Reply:
Oh, I probably understand what you mean. We remove the logic of the `closeBundle` hook in the `reporter` plugin, and then process it in `build.ts`, right?

Reply:
I don't know how useful the new line with "x Build failed in ms" is. At least if it is in red it may take away attention from the next lines
<img width="637" alt="image" src="https://github.com/vitejs/vite/assets/583075/a56ad002-6ad7-4604-b88d-5c84a2939297">

Maybe we could make only the x red? 
<img width="637" alt="image" src="https://github.com/vitejs/vite/assets/583075/1fea5024-d52e-4aff-8a8a-465da2f49dfe">

If there isn't a good reason to have this line, I vote to avoid it though

Reply:
Looks good, I'm leaning toward just setting the `x` to red.

Reply:
Ok, let's try that then 👍🏼